ivy/functional/ivy/experimental/linear_algebra.py

```diff
@@ -1666,3 +1666,90 @@ def dot(
     ivy.array([[-15.28]])
     """
     return current_backend(a, b).dot(a, b, out=out)
+
+
+@handle_exceptions
+@handle_nestable
+@handle_array_like_without_promotion
+@inputs_to_ivy_arrays
+@handle_array_function
+@handle_device_shifting
+def general_inner_product(
+    a: Union[ivy.Array, ivy.NativeArray],
+    b: Union[ivy.Array, ivy.NativeArray],
+    n_modes: Optional[int] = None,
+    /,
+    *,
+    out: Optional[ivy.Array] = None,
+) -> Union[ivy.Array, float]:
```

Comment:
I think we will always return an ivy.Array even if it just contains a single element.
```suggestion
) -> Union[ivy.Array]:
```